google-http-client-gson/src/main/java/com/google/api/client/json/gson/GsonFactory.java

```diff
@@ -90,4 +92,22 @@ public JsonGenerator createJsonGenerator(OutputStream out, Charset enc) {
   public JsonGenerator createJsonGenerator(Writer writer) {
     return new GsonGenerator(this, new JsonWriter(writer));
   }
+
+  /** Returns true if it is lenient to input JSON value. */
+  boolean getReadLeniency() {
+    return readLeniency;
+  }
+
+  private static GsonFactory newInstance(GsonFactory gsonFactory) {
+    GsonFactory copy = new GsonFactory();
+    copy.readLeniency = gsonFactory.readLeniency;
+    return copy;
+  }
+
+  /** Returns a copy of GsonFactory instance which is lenient when reading JSON value. */
+  public GsonFactory withReadLeniency() {
+    GsonFactory copy = newInstance(this);
```

Comment:
If we're supporting the subclassing use-case, I don't see how this works. Is the intent that subclasses are required to override this function? (If so, we're introducing a backwardly incompatible change to the subclass contract.)

As an example:
```java
// Logs all method invocations.
public LoggingGsonFactory extends GsonFactory { /* ... */ }

// In some 'main'-type class of the application...
public static GsonFactory createGsonFactory() { return new LoggingGsonFactory(); }

// Then elsewhere...
private GsonFactory gsonFactory = createGsonFactory().withReadLeniency(); 
// Oops! No more logging.
```

Reply:
Good point. Then subclasses needs to override newInstance. I'm making newInstance protected.

Reply:
But now it's static.

Reply:
Consider avoiding the copy problem altogether.
```java
private boolean isLenient;
public GsonFactory() { this(false); }
public GsonFactory(boolean isLenient) { this.isLenient = isLenient; }
```

Reply:
The boolean flag in the constructor is not good. It's ambiguous, especially when we want to introduce leniency in writing JSON.

Reply:
Added `   * <p>Subclasses should not call this method. Set {@code readLeniency} field to {@code true}
   * instead.`.

Reply:
"Subclasses should not call this method" is fine if you know you're dealing with a subclass. The point of polymorphism is that you shouldn't care.

There's other ways to take in a setting besides using a boolean if you feel it goes against best practices.

```java
public enum Leniency {
  LENIENT, STRICT
}
private final Leniency leniency;
public GsonFactory() { this(Leniency.STRICT); }
public GsonFactory(Leniency leniency) { this.leniency = Objects.requireNonNull(leniency); }
```
or a Builder, or a Settings object parameter.

Reply:
As talked, we may want to add more parameters (such as writeLeniency) in future and having different contstructors for different parameter is not good.

I'm bringing Builder class.